dirsrvtests/tests/suites/basic/basic_test.py

```diff
@@ -2186,6 +2194,194 @@ def test_dscreate_with_different_rdn(dscreate_test_rdn_value):
             assert True
 
 
+@pytest.fixture(scope="module")
+def dscreate_instance(request):
```

Comment:
I see that we have couple of similar tests in basic test suite that use this way of creating the instance. What is the reason for this? Can't we use the ones from topologies.py or create a new topology with the required settings?

Reply:
One of the reason is that theses tests use some non default values typically systemd = False (to disable systemctl) 
The sad thing is that our test topology framework lack the ability to specify arbitrary template parameters ( or dse.ldif config parameters that requires restart ) but anyway I think that such a change should be done as part of issue #3082 (the removal of pytest dependency on lib38)

Reply:
> One of the reason is that theses tests use some non default values typically systemd = False (to disable systemctl) The sad thing is that our test topology framework lack the ability to specify arbitrary template parameters ( or dse.ldif config parameters that requires restart ) but anyway I think that such a change should be done as part of issue #3082 (the removal of pytest dependency on lib38)

That was an extremely intentional design choice. It was to force growing lib389 to support the full surface area of our server config, so that it could become the cli tools that we have today. It also meant that the surface area of the admin tool internals is being tested incidentally during these integration tests. And it shows the "process" an admin would go through to make these changes, and how that would happen in reality. 

The other part was that dse.ldif was getting a bit "out of control" template wise. I wanted to get things to work in containers and that meant we had to support in place migrations and upgrades, so any value that existed in dse.ldif could be "removed" and a default from libglobs used. It could never be perfect, but it's significantly helped reduce complexity in that area for users.

Reply:
- I think you answered to my second point (why not having a phase allowing to tweak the dse.ldif
    before the initial startup). Ok this argument makes sense even if sometime it also makes the test case very complex (for example having to export the db, remove the db files and reimport when changing some db parameter like the page size) in some case.

  - But my main point is that right now it is cumbersome to use lib389 to create an instance 
   without explicitly using dscreate subprocesses.
    Creating an instance that is named "foo" is simple with dscreate, it should also be simple with direct call to lib389 too.  IMHO the upper level function that mimics dscreate (but providing the data from a dict instead of the dscreate template file) is just missing.